### ISSUE TITLE
Patching should clear out previous root element content

### ISSUE THREAD

**Issue by amirouche** [snabbdom/snabbdom#167]
Given the following html:

``` html
  <div id="app">should disappear</div>
```

And the following js code

``` javascript
var patch = snabbdom.init([
    snabbdom_style,
    snabbdom_class,
    snabbdom_props,
    snabbdom_attributes,
    snabbdom_eventlisteners
]);

var vnode = h('div', [" but it", " doesn't"]);
var container = document.getElementById('app');

patch(container, vnode);
```

`patch` will generate the following html:

``` html
<div id="app">should disappear but it doesn't</div>
```

One can work around this by setting an empty id for instance:

``` javascript
var vnode = h('div#', [" but it", " doesn't"]);
```

**Comment by amirouche** [comment]
Here is a jsbin http://jsbin.com/sovesitade/1/edit?html,output

**Comment by amirouche** [comment]
Also the `id` is wrong

**Comment by amirouche** [comment]
It's not possible to reproduce in the test suite. The output of patch is correct but the DOM is not.

**Comment by amirouche** [comment]
Here is another bug with similar behavior. With the same html as above, patching the div with the following `vnode`:

```javascript
var vnode = h('div', {attrs: {id: "app"}}, [" but it", " doesn't"]);
```

Won't work.

But the same call with `div#` seem to give the good result.

**Comment by WorldMaker** [comment]
This seems to be a regression between 0.5.0 and 0.5.3, if that helps narrow down the cause.

**Comment by paldepind** [comment]
Thank you for reporting this @amirouche. Also thanks to @WorldMaker for the additional information.

This is probably related to some recent changes to how `patch` with an element as it's first argument is done.

**Comment by AlexGalays** [comment]
Could @staltz help perhaps, since the regression appears to come from that change?

**Comment by staltz** [comment]
I'll look into it, but probably after the TypeScript migration.

**Comment by staltz** [comment]
Done and released in v0.6.4

**Comment by avesus** [comment]
Thank you!
